Automattic/wp-calypso
# Partner Portal: implement assign multiple licenses flow

#### Proposed Changes

This PR implements assigning multiple licenses flow, where the issue license process begins from the dashboard. If you visit directly the `/issue-license` page, this might not work as expected because we don't yet have the ID of the site.

This patch is built on top of https://github.com/Automattic/wp-calypso/issues/69183, which is the mutation that handles issuing of multiple licenses. In theory, it should completely replace `useLicenseIssuing`, but I have decided to keep the latter to avoid breaking the existing flows on production.

This PR includes:
- Transform radio buttons to checkboxes to avoid multiple product selection
- Display correct feedback notices when the user issues multiple licenses

This PR does not include the following:
- Incompatible product validations

**Prerequisite**
- Make sure your partner account is set to an agency and your partner key's billable type is set to Stripe.


#### Testing Instructions

- Checkout this branch locally
- Run `yarn start-jetpack-cloud`
- Create a few news JN sites
- Make sure your partner account is set to an agency and your partner key's billable type is set to non-billable.
- Apply D91613-code on your sandbox for easier testing
- Connect Jetpack
- Visit http://jetpack.cloud.localhost:3000/dashboard
- Select backup and scan for your new JN site
- On the page for issuing licenses, make sure that you can select multiple licenses and deselect them
- Verify that both licenses were issued and the feedback notice is correct
- Visit http://jetpack.cloud.localhost:3000/dashboard
- Select the backup license
- On the page for issuing licenses, make sure that you select bundle, and not a product
- Verify that the bundle license is issued and that the feedback notices are correct

**Important**
- Visit the Jetpack Cloud live link (horizon environment) in this PR and verify this PR isn't causing any regression. You must test various flows there, including setting your partner account billing type to Stripe. The basic flows are listed here: pdpAdu-v0-p2

**Screenshots**

Multiple products(Success)

<img width="1454" alt="Screenshot 2022-11-08 at 5 01 13 PM" src="https://user-images.githubusercontent.com/10586875/200553696-61607280-8cb6-4175-b0ff-c57edc6fd032.png">

Single product(Success)

<img width="1454" alt="Screenshot 2022-11-08 at 5 02 12 PM" src="https://user-images.githubusercontent.com/10586875/200553721-21979dea-bb73-4d94-9f9b-fb0b9f597911.png">

Multiple products(Failure)

<img width="1454" alt="Screenshot 2022-11-08 at 5 01 38 PM" src="https://user-images.githubusercontent.com/10586875/200553710-9fee7eef-8154-4932-ab99-28ac4875aa3c.png">

Single Product(Failure)

<img width="1454" alt="Screenshot 2022-11-08 at 5 01 53 PM" src="https://user-images.githubusercontent.com/10586875/200553717-a0c4b174-4436-4b3f-8ca8-d2f6c2245551.png">



#### Pre-merge Checklist



Related to 1203126240279377-as-1203126240279405 & 1203126240279377-as-1203188906065242
